File: fidanys-server/src/main/java/com/fidanlik/fidanysserver/goodsreceipt/model/GoodsReceipt.java
package com.fidanlik.fidanysserver.goodsreceipt.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Document(collection = "goodsReceipts")
public class GoodsReceipt {
    @Id
    private String id;
    private String receiptNumber;
    // Eski supplierId alanı kaldırıldı veya ismi değiştirildi, çünkü artık daha genel bir sourceId var.
    // private String supplierId;
    private String warehouseId;
    private List<GoodsReceiptItem> items;
    private BigDecimal totalValue;
    private GoodsReceiptStatus status;
    private String userId;
    private LocalDateTime receiptDate;
    private String tenantId;

    // YENİ EKLENEN ALANLAR
    private SourceType sourceType; // Mal girişinin kaynağı (Tedarikçi mi, Üretim Partisi mi?)
    private String sourceId; // sourceType'a göre ya supplierId ya da productionBatchId

    // Giderin hangi ödeme ile yapıldığını belirtir -> BU ALAN SANIRIM YANLIŞLIKLA GELMİŞ, KALDIRILDI
    // private String paymentId;

    // YENİ ENUM: Mal Giriş Kaynağı
    public enum SourceType {
        SUPPLIER,           // Tedarikçiden gelen mal
        PRODUCTION_BATCH    // Kendi üretim partisinden gelen mal (hasat)
    }

    public enum GoodsReceiptStatus {
        COMPLETED,
        CANCELED
    }
}

File: client/src/types/goods-receipt.ts
// client/src/types/goods-receipt.ts

import type { Plant } from './plant'; // Eğer GoodsReceiptItem'da Plant detaylarına ihtiyaç duyulursa

// GoodsReceiptItem tipi (server tarafındaki model ile uyumlu)
export interface ReceiptItemDto {
  plantId: string;
  quantity: number;
  unitCost: number; // Fidanın birim maliyeti (alış fiyatı veya üretim birim maliyeti)
  // Bu alana ihtiyacımız kalmadı, GoodsReceiptRequest içindeki sourceType tarafından yönetiliyor.
  // isCommercial?: boolean;
  // productionBatchId?: string; // GoodsReceiptRequest'in sourceId'si olarak taşındı
}

// GoodsReceipt tipi (server tarafındaki model ile uyumlu)
export interface GoodsReceipt {
  id: string;
  receiptNumber: string;
  // Eski supplierId alanı yerine sourceType ve sourceId kullanıldı
  // supplierId?: string;
  warehouseId: string;
  items: ReceiptItemDto[];
  totalValue: number; // BigDecimal yerine number kullanıldı (client tarafı için)
  status: 'COMPLETED' | 'CANCELED'; // Enum yerine union type
  userId: string;
  receiptDate: string; // LocalDateTime yerine ISO string
  tenantId: string;
  sourceType: 'SUPPLIER' | 'PRODUCTION_BATCH'; // Mal girişinin kaynağı
  sourceId: string; // sourceType'a göre supplierId veya productionBatchId
}